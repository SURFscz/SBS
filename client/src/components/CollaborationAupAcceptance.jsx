import React from "react";
import I18n from "../locale/I18n";
import "./CollaborationAupAcceptance.scss";
import "./welcome/welcome.scss";
import CheckBox from "./CheckBox";
import Logo from "./redesign/Logo";
import {isEmpty} from "../utils/Utils";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";

export default function CollaborationAupAcceptance({
                                                       services,
                                                       disabled,
                                                       serviceEmails,
                                                       setDisabled,
                                                       allServiceAupsAgreedOn,
                                                       children
                                                   }) {
    const renderServiceAup = service => {
        const hasEmails = !isEmpty(serviceEmails) && !isEmpty(serviceEmails[service.id]);
        const mails = hasEmails ? encodeURI(serviceEmails[service.id].join(",")) : null;

        return (
            <div className="service-section" key={service.id}>
                <table>
                    <thead/>
                    <tbody>
                    <tr>
                        <td className="logo">
                            {service.logo && <Logo src={service.logo} alt={service.name}/>}
                        </td>
                        <td className="service-detail">
                            <div className="service">
                                <span className="service-name">{service.name}</span>
                                <div className="policies">
                                    <div className="policy">
                                        {service.accepted_user_policy ?
                                            <a href={service.accepted_user_policy}
                                               rel="noopener noreferrer"
                                               target="_blank">{I18n.t("service.accepted_user_policy")}</a> :
                                            <span className="no-link">{I18n.t("aup.service.noAup")}</span>
                                        }
                                    </div>
                                    <div className="policy">
                                        {service.privacy_policy ?
                                            <a href={service.privacy_policy}
                                               rel="noopener noreferrer"
                                               target="_blank">{I18n.t("service.privacy_policy")}</a> :
                                            <span className="no-link">{I18n.t("aup.service.noPrivacyPolicy")}</span>
                                        }
                                    </div>
                                    <div className="policy contact">
                                        {mails ? <a href={`mailto:${mails}`} rel="noopener noreferrer">
                                            <FontAwesomeIcon
                                                icon="envelope"/><span>{I18n.t("aup.service.contact")}</span>
                                        </a> : <span className="no-link">{I18n.t("aup.service.noContact")}</span>}
                                    </div>
                                </div>
                            </div>
                        </td>
                    </tr>
                    </tbody>
                </table>
            </div>
        );
    }
    const servicesWithAups = services.filter(service => !isEmpty(service.accepted_user_policy));
    const singleServiceWithAup = servicesWithAups.length === 1;
    return (
        <div className="mod-aup-services">
            {children}
            {services.length > 0 && <div className="services">
                {services.map(service => renderServiceAup(service))}
            </div>}
            {services.length === 0 && <div className="services">
                <span>{I18n.t("welcomeDialog.noServices")}</span>
            </div>}
            {(servicesWithAups.length > 0 && !allServiceAupsAgreedOn) &&
                <div className="terms">
                    <p className="aup-info">
                        {I18n.t(`aup.service.${singleServiceWithAup ? "singleInfo" : "multipleInfo"}`)}
                    </p>
                    <CheckBox name="aup"
                              value={!disabled}
                              info={I18n.t(`aup.service.${singleServiceWithAup ? "singleCheck" : "multipleCheck"}`)}
                              onChange={() => setDisabled(!disabled)}/>
                </div>}

            {(servicesWithAups.length > 0 && allServiceAupsAgreedOn) &&
                <div className="terms">
                    <p className="aup-info">
                        {I18n.t(`aup.service.${singleServiceWithAup ? "singleInfoAccepted" : "multipleInfoAccepted"}`)}
                    </p>
                </div>}
        </div>);

}
