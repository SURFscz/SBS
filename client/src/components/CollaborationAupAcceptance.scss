@import "../stylesheets/vars.scss";

div.mod-aup-services {
    div.service-section {
        display: flex;
        padding: 7px 10px;
        border-bottom: 1px solid $lighter-grey;
        align-items: center;
        width: 100%;
        min-height: 56px;
        margin-bottom: 15px;

        &:first-child {
            border-top: 3px solid $lighter-grey;
        }

        img {
            border-radius: 4px;
            max-width: 76px;
        }

        div.border-right {
            margin-right: 25px;
            padding-right: 25px;
            border-right: 1px solid $dark-grey;
        }

        table {
            width: 100%;

            td.logo {
                width: 85px;
            }

            td.service-detail {
                width: 90%;

                div.service {
                    display: flex;
                    flex-direction: column;
                    margin-left: 15px;

                    span.service-name {
                        font-weight: bold;
                        margin-bottom: 6px;
                    }
                }

                div.policies {
                    display: flex;
                }

                div.contact {
                    display: flex;
                    align-content: center;
                    svg {
                        margin-right: 10px;
                    }
                }
            }

        }

        span.no-link {
            color: $darker-grey;
            font-style: italic;
        }
    }

    div.terms {
        margin: 25px 0 0 0;
        background-color: $background;
        padding: 15px;
    }


}