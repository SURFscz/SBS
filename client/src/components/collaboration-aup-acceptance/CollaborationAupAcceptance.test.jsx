import React from "react";
import {render} from "@testing-library/react";
import CollaborationAupAcceptance from "./CollaborationAupAcceptance";

jest.mock("../../locale/I18n", () => ({
    t: key => key
}));
jest.mock("../checkbox/CheckBox", () => props => (
    <div data-testid="checkbox">{JSON.stringify(props)}</div>
));
jest.mock("../_redesign/Logo", () => props => (
    <img data-testid="logo" {...props} />
));
jest.mock("../../utils/Utils", () => ({
    isEmpty: obj => !obj || (typeof obj === "object" && Object.keys(obj).length === 0)
}));
jest.mock("@fortawesome/react-fontawesome", () => ({
    FontAwesomeIcon: () => <span data-testid="icon"/>
}));

const services = [
    {
        id: 1,
        name: "Service One",
        logo: "logo1.png",
        accepted_user_policy: "https://aup.example.com",
        privacy_policy: "https://privacy.example.com"
    }
];

const serviceEmails = {
    1: ["support@example.com"]
};

describe("CollaborationAupAcceptance", () => {
    it("matches snapshot", () => {
        const {asFragment} = render(
            <CollaborationAupAcceptance
                services={services}
                disabled={true}
                serviceEmails={serviceEmails}
                setDisabled={jest.fn()}
                allServiceAupsAgreedOn={false}
            >
                <div>Child content</div>
            </CollaborationAupAcceptance>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
