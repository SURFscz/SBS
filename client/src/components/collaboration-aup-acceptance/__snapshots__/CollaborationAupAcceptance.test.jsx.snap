// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollaborationAupAcceptance matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mod-aup-services"
  >
    <div>
      Child content
    </div>
    <div
      class="services"
    >
      <div
        class="service-section"
      >
        <table>
          <thead />
          <tbody>
            <tr>
              <td
                class="logo"
              >
                <img
                  alt="Service One"
                  data-testid="logo"
                  src="logo1.png"
                />
              </td>
              <td
                class="service-detail"
              >
                <div
                  class="service"
                >
                  <span
                    class="service-name"
                  >
                    Service One
                  </span>
                  <div
                    class="policies"
                  >
                    <div
                      class="policy"
                    >
                      <a
                        href="https://aup.example.com"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        service.accepted_user_policy
                      </a>
                    </div>
                    <div
                      class="policy"
                    >
                      <a
                        href="https://privacy.example.com"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        service.privacy_policy
                      </a>
                    </div>
                    <div
                      class="policy contact"
                    >
                      <a
                        href="mailto:support@example.com"
                        rel="noopener noreferrer"
                      >
                        <span
                          data-testid="icon"
                        />
                        <span>
                          aup.service.contact
                        </span>
                      </a>
                    </div>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="terms"
    >
      <p
        class="aup-info"
      >
        aup.service.singleInfo
      </p>
      <div
        data-testid="checkbox"
      >
        {"name":"aup","value":false,"info":"aup.service.singleCheck"}
      </div>
    </div>
  </div>
</DocumentFragment>
`;
